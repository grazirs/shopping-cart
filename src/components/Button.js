import styled from "styled-components";

export const Button = styled.button `
  background-color: ${({theme}) => theme.colors.blue};
  min-width: 64px;
  text-transform: uppercase;
  border-radius: ${({theme})=> `${theme.radii[1]}px`};
  padding: ${({theme}) => `${theme.space[2]}px`};
  border: none;
  color: ${({theme}) => theme.colors.white};
  cursor: pointer;
  text-align: center;
  &:disabled {
    background-color: ${({theme}) => theme.colors.grey};
    color: ${({theme}) => theme.colors.darkGrey};
  }
`
