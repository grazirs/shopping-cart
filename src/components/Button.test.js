import renderer from 'react-test-renderer';
import { Button } from "./Button";
import theme from '../theme';
import { ThemeProvider } from 'styled-components';

describe("<Button/>", () => {
  it("renders Button component correctly", () => {
    const button = renderer.create(<ThemeProvider theme={theme}><Button>Button</Button></ThemeProvider>).toJSON();
    expect(button).toMatchSnapshot();
  })
});
