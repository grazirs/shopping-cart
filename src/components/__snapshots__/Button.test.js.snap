// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> renders Button component correctly 1`] = `
<button
  className="Button-sc-2h6cpf-0 dgfxzp"
>
  Button
</button>
`;
